document config for sniffing compressed files, enable by default

martenson opened the pull request:
Me and @natefoo think that it is time to switch the default of this config option to True. At least for Main.

In addition I suspect there is a bug in data_source tools that prevent/avoid uncompressing uploaded archives and thus causing issues like https://github.com/galaxyproject/galaxy/issues/6334.

ping @jmchilton

martenson committed:
document the sniff_compressed_dynamic_datatypes_default config and include in sample

natefoo commented:
Just to walk it back slightly - I think we should do this, but I think we ought to check the proportion of jobs over time running tools with fastq inputs and see how many of those are tools that support native `fastqsanger.gz` inputs first. If the majority of such jobs are still using non-`fastqsanger.gz`-supporting tools, this would probably be a net loss for usegalaxy.org (but we should work quickly to get any remaining tools updated to supporting `fastqsanger.gz`).

martenson commented:
It could technically be a short run net loss (more space consumed because of the compressed archive presence in history after implicit conversion) but it is the path forward we want to promote I believe. 

It will motivate tool authors to write tools that accept compressed datatypes as first-class citizens and also improve our storage efficiency in the long run. 

>If the majority of such jobs are still using non-fastqsanger.gz-supporting tools

I wouldn't give this ratio too much importance, since we are now being inefficient (on Main) by unpacking all compressed formats during upload that are not specifically set to compressed datatype (and setting compressed datatype is rarely used I believe). By stopping this practice (with enabling sniffing of compressed datatypes) we will save space in cases that will never run implicit conversion afterwards. My guess is that this saved space outweighs the extra space consumed by preservation of the compressed copies after implicit conversion (for the analyses that use it).

jmchilton commented on lib/galaxy/datatypes/registry.py:
```diff
@@ -141,7 +141,7 @@ def __import_module(full_path, datatype_module, datatype_class_name):
                 sniff_compressed_types = galaxy.util.string_as_bool_or_none(elem.get("sniff_compressed_types", "None"))
                 if sniff_compressed_types is None:
-                    sniff_compressed_types = getattr(self.config, "sniff_compressed_dynamic_datatypes_default", True)
+                    sniff_compressed_types = getattr(self.config, "sniff_compressed_dynamic_datatypes", True)
```
This can be configured/overridden on a per-datatype basis in the datatypes_conf.xml - so I think the old config value name is more appropriate.

martenson replied:
I removed the renaming and added the note about configuration to the description. Thanks for pointing this out.

jmchilton commented:
Did you mess up your gitfu here - I don't see the original changes anymore. I'd use git reflog to find the old head if you no longer have them in a named branch.

martenson commented:
@jmchilton this is now a documentation PR only. I see it fine I think.

martenson committed:
switch the default to sniff compressed datatypes

martenson commented:
I have added a commit that switches the default for sniffing compressed to `True`.

jmchilton commented:
I was super confused and remember the original PR being bigger and switching the default, I'm... old I guess.

Thanks for switching the default - I think all contemporary tools have been written with compressed files in mind and this will definitely save us space - but regardless you points about the ratio and being future facing are absolutely true regardless of that debate.

nsoranzo commented:
@martenson API test failures seem legit.

martenson committed:
adjust test definitions for compressed handling

so it tests the new defaults

martenson commented:
I have added commit with adjustments to the compressed upload API tests. I see strange results locally so let's see what jenkins sees.

mvdbeek committed:
Use uses_test_history for selected upload tests

martenson committed:
Merge pull request #29 from mvdbeek/martenson-rename-and-document-config-value

Use uses_test_history for selected upload tests

martenson committed:
fix tests for the bed.gz dtaatype

compressed beds are not part of the datatypes_conf.xml.sample

martenson committed:
Merge branch 'rename-and-document-config-value' of github.com:martenson/galaxy into rename-and-document-config-value

martenson committed:
add back the unstable flags for bed.gz datasets

I expect the datatypes_conf.xml bed compress default will change in near future

martenson commented:
Tests now pass for me locally. Hopefully they make sense.

jennaj commented:
This change will help for issues with fastqsanger data, however, if the data is fastqcssanger, probably not. That type of data is being sniffed incorrectly in two ways (.gz compression not recognized **and** base fastq type is wrong) -- not sure at what step that is done. 

**In short, EBI SRA "Send to Galaxy" data cannot be assumed to be fastqsanger.** Example and details here: https://github.com/galaxyproject/galaxy/issues/6334#issuecomment-419205445

martenson commented:
@jennaj I don't think this affects the current PR. Please provide examples of these wrongly sniffed `fastqcssanger` files and create a new issue for it.
